code/drasil-gool/GOOL/Drasil/State.hs

```diff
@@ -494,8 +494,8 @@ addIter st = over iterators ([st]++)
 getIter :: MS [String]
 getIter = gets (^. iterators)
 
-resetIter :: MethodState -> MethodState
-resetIter = set iterators []
+resetIter :: String -> MethodState -> MethodState
+resetIter st = over iterators (delete st)
```

Comment:
I had to change the resetIter function to this because I realized that my previous changes could lead to bugs with nested for each loops. If you want to know what bug is, then in a nutshell: Previously the code written within the outer for loop but after the inner for loop is closed would have produced errors because ```resetIter``` removed all Iter variables (including the outer one).